docs/en/install-upgrade/installing-stack-demo-secure.asciidoc

```diff
@@ -0,0 +1,896 @@
++
+["source","shell"]
+----
+ls -lR /usr/share/elasticsearch/{elasticsearch,kibana}
+----
++
+["source","shell"]
+----
+elasticsearch:
+total 0
+drwxr-xr-x. 2 root root 56 Dec 12 19:13 mynode-es1
+drwxr-xr-x. 2 root root 72 Dec 12 19:04 mynode-es2
+drwxr-xr-x. 2 root root 72 Dec 12 19:04 mynode-es3
+
+elasticsearch/mynode-es1:
+total 8
+-rw-r--r--. 1 root root 1365 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root  845 Dec 12 19:04 sample-elasticsearch.yml
+
+elasticsearch/mynode-es2:
+total 12
+-rw-r--r--. 1 root root 3652 Dec 12 19:04 http.p12
+-rw-r--r--. 1 root root 1365 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root  845 Dec 12 19:04 sample-elasticsearch.yml
+
+elasticsearch/mynode-es3:
+total 12
+-rw-r--r--. 1 root root 3652 Dec 12 19:04 http.p12
+-rw-r--r--. 1 root root 1365 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root  845 Dec 12 19:04 sample-elasticsearch.yml
+
+kibana:
+total 12
+-rw-r--r--. 1 root root 1200 Dec 12 19:04 elasticsearch-ca.pem
+-rw-r--r--. 1 root root 1306 Dec 12 19:04 README.txt
+-rw-r--r--. 1 root root 1052 Dec 12 19:04 sample-kibana.yml
+----
+
+. Replace your existing keystore with the new keystore. The location of your certificate directory may be different than what is shown here, depending on the installation method you chose.
++
+Run the `mv` command, replacing `<es1-hostname>` with the hostname of your initial {es} node:
++
+["source","shell"]
+----
+sudo mv /usr/share/elasticsearch/elasticsearch/<es1-hostname>/http.p12 /etc/elasticsearch/certs/
+----
+
+. Because this is a new keystore, the {es} configuration file needs to be updated with the path to its location. Open `/etc/elasticsearch/elasticsearch.yml` and update the HTTP SSL settings with the new path:
++
+["source","yaml"]
+----
+xpack.security.http.ssl:
+  enabled: true
+  #keystore.path: certs/http.p12
+  keystore.path: /etc/elasticsearch/certs/http.p12
+----
+
+. Since you also generated a new keystore password, the {es} keystore needs to be updated as well. From the {es} installation directory, first remove the existing HTTP keystore entry:
++
+["source","shell"]
+----
+sudo /usr/share/elasticsearch/bin/elasticsearch-keystore remove xpack.security.http.ssl.keystore.secure_password
+----
+
+. Add the updated HTTP keystore password, using the password you generated for this keystore:
++
+["source","shell"]
+----
+sudo /usr/share/elasticsearch/bin/elasticsearch-keystore add xpack.security.http.ssl.keystore.secure_password
+----
+
+. Because we've added files to the {es} configuration directory during this tutorial, we need to ensure that the permissions and ownership are correct before restarting {es}.
+
+.. Change the files to be owned by `root:elasticsearch`:
++
+["source","shell"]
+----
+sudo chown -R root:elasticsearch /etc/elasticsearch/certs/
+----
+
+.. Set the files in `/etc/elasticsearch/certs` to have read and write permissions by the owner (`root`) and read permission by the `elastic` user:
++
+["source","shell"]
+----
+chmod 640 *
```

Comment:
I don't want anyone to blame us for hosing their system when they set permissions incorrectly in the wrong directory, so I think we should be explicit here:

`sudo chmod 640 /etc/elasticsearch/certs/elastic-certificates.p12`
`sudo chmod 640 /etc/elasticsearch/certs/elastic-stack-ca.p12`
`sudo chmod 640 /etc/elasticsearch/certs/http_ca.crt`
`sudo chmod 640 /etc/elasticsearch/certs/http.p12`